Quaver.Shared/Screens/Gameplay/UI/PauseScreen.cs

```diff
@@ -131,6 +106,46 @@ public PauseScreen(GameplayScreen screen)
             Quit.Size = new ScalableVector2(Quit.Image.Width, Quit.Image.Height);
         }
 
+        /// <summary>
+        ///     Called when Continue Button gets selected
+        /// </summary>
+        private void InitiateContinue()
+        {
+            if (!Screen.IsPaused)
+                return;
+            GameBase.Game.GlobalUserInterface.Cursor.Alpha = 0;
+            Screen.Pause();
+        }
+
+        /// <summary>
+        ///     Called when Retry Button gets selected
+        /// </summary>
+        private void InitiateRetry()
+        {
+            if (!Screen.IsPaused)
+                return;
+            GameBase.Game.GlobalUserInterface.Cursor.Alpha = 0;
+            SkinManager.Skin.SoundRetry.CreateChannel().Play();
+            QuaverScreenManager.ChangeScreen(new GameplayScreen(Screen.Map, Screen.MapHash, Screen.LocalScores));
```

Comment:
It would be worth implementing a method in `GameplayScreen` specifically for retrying. With what you have now, you're assuming the user wants to retry to *play* the map and it doesn't handle the case of when they're watching a replay and wants to restart it.

See: [GameplayScreen.cs#L593](https://github.com/Quaver/Quaver/blob/develop/Quaver.Shared/Screens/Gameplay/GameplayScreen.cs#L593)